edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -2743,18 +2743,29 @@ public String getAverageCourseGrade(String gradebookUid) {
 	 * 
 	 * @see GradebookService.updateAssignmentOrder(java.lang.String gradebookUid, java.lang.Long assignmentId, java.lang.Integer order)
 	 */
-	public void updateAssignmentOrder(final String gradebookUid, final Long assignmentId, final Integer order) {
+	public void updateAssignmentOrder(final String gradebookUid, final Long assignmentId, Integer order) {
 		
 		if (!getAuthz().isUserAbleToEditAssessments(gradebookUid)) {
 			log.error("AUTHORIZATION FAILURE: User " + getUserUid() + " in gradebook " + gradebookUid + " attempted to change the order of assignment " + assignmentId);
 			throw new SecurityException("You do not have permission to perform this operation");
 		}
+		
+		if(order == null) {
+			throw new IllegalArgumentException("Order cannot be null");
+		}
 	
 		final Long gradebookId = getGradebook(gradebookUid).getId();
 		
 		//get all assignments for this gradebook
 		List<Assignment> assignments = getAssignments(gradebookId, SortType.SORT_BY_SORTING, true);
 		
+		//adjust order to be within bounds
```

Comment:
This conditional can be replaced with something like this:

``//adjust order to be within bounds``
``order = Math.min ( Math.max ( 0 , order ) , assignments.size() );``

IMHO this is more compact and also readable, but is just my opinion :)